src/test/kotlin/uk/gov/dluhc/emsintegrationapi/database/repository/PostalVoteApplicationRepositoryIntegrationTest.kt

```diff
@@ -56,4 +57,47 @@ class PostalVoteApplicationRepositoryIntegrationTest : AbstractRepositoryIntegra
             }
         }
     }
+
+    @Test
+    fun `should return records by gsscode and record status order by created date`() {
+        // Given
+        val gssCode = "1234"
+        val approvalDetailsValid = buildApprovalDetailsEntity(gssCode = gssCode)
+        val approvalDetailsInvalid = buildApprovalDetailsEntity(gssCode = "5678")
+        val listOfApplications =
+            IntStream.rangeClosed(1, 30).mapToObj {
+                val approvalDetails = if (it % 2 == 0) {
+                    approvalDetailsValid
+                } else {
+                    approvalDetailsInvalid
+                }
+                buildPostalVoteApplication(applicationId = it.toString(), approvalDetails = approvalDetails)
+            }.toList()
+
+        postalVoteApplicationRepository.saveAllAndFlush(listOfApplications)
+
+        // When
+        val applicationsReceived =
+            postalVoteApplicationRepository.findByApprovalDetailsGssCodeAndStatusOrderByDateCreated(
+                gssCode,
+                RecordStatus.RECEIVED,
+                Pageable.ofSize(10)
+            )
+
+        // That
+        val size = applicationsReceived.size
+        assertThat(size).isLessThan(11)
```

Comment:
Can we be more specific in the number of applications we are expecting back here?